AK/URL.cpp

```diff
@@ -389,7 +389,7 @@ bool URL::equals(URL const& other, ExcludeFragment exclude_fragments) const
     return serialize(exclude_fragments) == other.serialize(exclude_fragments);
 }
 
-DeprecatedString URL::basename() const
+Optional<DeprecatedString const&> URL::basename() const
```

Comment:
This feels like a slightly frivolous footgun, since the URL could be mutated and this reference would go stale.
I think we can allow ourselves to pay the cost of refcounting by copying the `DeprecatedString` here.
Keep in mind that copying a string by value is just refcounting churn, not allocation.

Also, typo in commit: covert -> convert.

Reply:
I've changed this function to `decoded_path_basename()` which I think now (which I think it what all its callers want).  Which means it may allocate now (but probably not due to the check in `percent_decode()`).